src/utils/url.mjs

```diff
@@ -0,0 +1,19 @@
+import { BASE_URL } from '../constants.mjs';
+
+/**
+ * Builds the url of a api doc entry.
+ *
+ * @param {ApiDocMetadataEntry} entry
+ * @param {boolean} [useHtml]
+ * @returns {URL}
+ */
+export const buildApiDocURL = (entry, useHtml = false) => {
+  const path = entry.api_doc_source.replace(/^doc\//, '/docs/latest/');
+
+  if (useHtml) {
+    const htmlPath = path.replace(/\.md$/, '.html');
```

Comment:
nit: no need to assign const htmlPath just return new URL();

Also instead of using new URL() on all these places, could you use URL.parse please?